src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -284,12 +284,16 @@ static std::tuple<std::vector<NodePtr>, std::vector<size_t>> ExtractExecutableNo
     std::unordered_map<size_t, size_t> graphIdToExecutableId;
     std::vector<NodePtr> executableGraphNodes;
     for (size_t i = 0; i < graphNodes.size(); i++) {
-        const auto& graphNode = graphNodes[i];
-        if ((!graphNode->isConstant() && graphNode->isExecutable()) ||  // non-constant executable or
-            (graphNode->isDynamicNode() &&
-             !one_of(graphNode->getType(), Type::Input, Type::Output))) {  // dynamic, except inputs / outputs
+        const auto& node = graphNodes[i];
+        const bool staticZeroDims = !node->isDynamicNode() && !node->isExecutable() && !node->isInPlace();
```

Comment:
If the only fact of interest is that there are empty tensors, it's better to call the check directly:
```suggestion
        const bool staticZeroDims = !node->isDynamicNode() && node->hasEmptyInputTensors();
```

Reply:
This was my initial implementation.
The problem is that some nodes are actually checking for the empty output tensors instead.
I have a prototype PR where the reasoning "isInPlace" and "empty tensors on data ports" are clearly distinguished which also helps to simplify a shape inference and update dynamic shapes procedures.